### DISCUSSION TITLE
Working on alternative setup based on LMN-3

### DISCUSSION THREAD

**Discussion by meirssv** [FundamentalFrequency/.github#69] [Show and tell]
Hello,

Your project attracted my interest and I started working on an alternative build. For this build I use a [Seeed Studio reTerminal](https://www.seeedstudio.com/ReTerminal-with-CM4-p-4904.html?queryID=c26237d06ee3367d41a85180a190dd82&objectID=4904&indexName=bazaar_retailer_products) with a Raspberry Pi CM4 module and integrated touchscreen. I got LMN-3 working on the reTerminal.

For the MIDI-controls I will also use a Teensy 4.1 (which is coming in this week) but I will use 128-led RGB-encoders, which I developed myself. ([See it in action](https://twitter.com/BelgoTrack/status/1438246912126226435)) These encoders use I2C, so technically I should be able to integrate these into the Surface Control-code of the Teensy firmware. These are two-way controllable, you can change the level by turning the encoder physically (duh), but you can also preset or modify the level using I2C-commands. I think it would be pretty easy to translate incoming MIDI CC commands into the Teensy to the I2C encoder.

My rgb-encoder defaults to a single LED presentation -as seen in the gif in the 'see it in action'-link above- , but it is also possible to light up all leds from '0' to the currently set level. However, this -uses a lot more current of course, driving 127 leds. Up to 400 mA with all leds active, the Teensy will not like this amount of power drawn, so at first I will only use the 'single led'-presentation while working out an alternative power source for the encoders.

Another worry I have is latency/delays in using I2C as the link between the Teensy (MIDI) and the encoders. I don't know if this will be fast enough for lets say real time CC MIDI messages to be sent out, but I hope to try this all out this week.

If there is interest in my RGB-encoders, please feel free to discuss it here.

**Comment by stonepreston** [comment]
Those encoders are super cool. Lemme know how things go
